app/src/main/java/chat/rocket/android/helper/RocketChatUserAvatar.kt

```diff
@@ -1,17 +1,91 @@
 package chat.rocket.android.helper
 
+import android.content.Context
+import android.graphics.Typeface
+import android.graphics.drawable.Drawable
+import com.amulyakhare.textdrawable.TextDrawable
 import java.net.URLEncoder
 
-class RocketChatUserAvatar(val hostname: String, val username: String) {
-
-    val imageUri: String
-        /** REMARK
-         * This is often a SVG image (see Rocket.Chat:server/startup/avatar.js).
-         */
-        get() {
-            return "https://" +
-                    hostname.replace("http://", "").replace("https://", "") +
-                    "/avatar/" +
-                    URLEncoder.encode(username, "UTF-8")
+object RocketChatUserAvatar {
+
+    /**
+     * Returns the user avatar URI.
+     *
+     * REMARK: This is often a SVG image (Rocket.Chat:server/startup/avatar.js).
+     *
+     * @param hostname The server's hostname.
+     * @param username The username.
+     * @return The user avatar URI.
+     * @see getImageFormat
+     */
+    fun getUri(hostname : String, username: String): String {
+        return "https://" +
+                hostname.replace("http://", "").replace("https://", "") +
+                "/avatar/" +
+                URLEncoder.encode(username, "UTF-8")
+    }
+
+    /**
+     * Returns the user avatar image format.
+     *
+     * @param uri The user avatar image URI to get the image format.
+     * @return The user avatar image format. Possible values are: "image/jpeg", "image/png" and "image/svg+xml".
+     * @see getUri
+     */
+    fun getImageFormat(uri: String): String {
+        return OkHttpHelper.getContentType(uri)
```

Comment:
We should have this information from the backend, each time we show an Avatar doing a HEAD (network operation) operation and a GET seems very expensive.